### DISCUSSION TITLE
Error in term_gene_heatmap function

### DISCUSSION THREAD

**Discussion by safrikDut** [egeulgen/pathfindR#85] [General]
Hi!
Thanks a lot for answering my last question! Now I have an error in the term_gene_heatmap function that I cannot figure out the source of error:

> term_gene_heatmap <- term_gene_heatmap(result_df = final_res, 
+                                        genes_df = DMSO_mock, 
+                                        num_terms = 20,
+                                        use_description = TRUE,
+                                        pin_name_path = "STRING",
+                                        low = "red",
+                                        mid = "black",
+                                        high = "green")

Error in input_testing(genes_df) : p values must all be numeric

I am using the same input file in genes_df as for run_PathfindR. I have also run enrichment_chart functions without any trouble.
Any help will be very much appreciated!

Thanks!

**Comment by egeulgen** [comment]
Hey @safrikDut,

Would you mind sharing the `DMSO_mock` and `final_res` as RDS files so that I can try to replicate the issue?

**Reply by safrikDut** [reply]
Hi!
Of course!
[files.zip](https://github.com/egeulgen/pathfindR/files/6609756/files.zip)

Thanks!

**Comment by egeulgen** [comment]
Hello again @safrikDut,
The issue was that `DMSO_mock` is a tibble (tbl_df), changing it into a regular data frame via `as.data.frame(DMSO_mock)` resolves the issue.
Another comment: you should not storing the heatmap object in a variable called `term_gene_heatmap` as this will override the function and you cannot use the function afterwards, I suggest simply using `g`.

PS I will probably implement a workaround so that pathfindR works with tibbles as well.
